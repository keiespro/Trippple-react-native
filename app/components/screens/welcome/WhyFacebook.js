import React from 'react';
import {
  Dimensions,
  Image,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native';

import colors from '../../../utils/colors';
import { MagicNumbers } from '../../../utils/DeviceConfig';

const DeviceHeight = Dimensions.get('window').height;
const DeviceWidth = Dimensions.get('window').width;

const WhyFacebook = ({kill}) => (
  <ScrollView style={{ width: DeviceWidth, height: DeviceHeight, backgroundColor: colors.outerSpace, padding: MagicNumbers.screenPadding}}>
    <View style={{flex: 1, flexDirection: 'column', justifyContent: 'space-around', height: DeviceHeight - 50}}>
      <Text style={[styles.allText, styles.titleText, {textAlign: 'center', marginTop: 20}]}>WHY FACEBOOK?</Text>
      <View>
        <Text style={[styles.allText, styles.titleText, {}]}>HIDE FROM YOUR FRIENDS</Text>
        <Text style={[styles.allText, {}]}>Facebook login makes it easier to hide you from Facebook friends and family</Text>
      </View>
      <View>
        <Text style={[styles.allText, styles.titleText, {}]}>NO FAKE USERS</Text>
        <Text style={[styles.allText, {}]}>Facebook does a pretty good job at purging fake users. Signing up with Facebook lowers the chances of fake users joining Trippple.</Text>
      </View>
      <View>
        <Text style={[styles.allText, {color: colors.steelGrey}]}>
          Note: We do not store inactive user information. Disabling your Tripple account will delete all of your Facebook information from Trippple.
        </Text>
      </View>
      <TouchableOpacity
        style={{backgroundColor: colors.transparent, alignSelf: 'center'}}
        onPress={() => { kill() }}
      >
        <Image source={require('../../../assets/close.png')} style={{width: 35, height: 35}} />
      </TouchableOpacity>
    </View>
  </ScrollView>
)


const styles = StyleSheet.create({
  allText: {
    color: colors.white,
    fontSize: 18,
    fontFamily: 'omnes',
    textAlign: 'left',
  },
  titleText: {
    fontSize: 16,
    fontFamily: 'montserrat',
    fontWeight: '800',
  },
});

export default WhyFacebook
