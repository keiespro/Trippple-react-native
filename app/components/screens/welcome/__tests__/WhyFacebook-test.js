import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';

import WhyFacebook from '../WhyFacebook';

test('renders correctly', () => {
  const tree = renderer.create(
    <WhyFacebook />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
