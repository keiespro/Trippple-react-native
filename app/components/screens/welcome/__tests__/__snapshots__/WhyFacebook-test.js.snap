// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "rgb(44,57,65)",
      "height": 1334,
      "padding": 50,
      "width": 750,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "height": 1284,
          "justifyContent": "space-around",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(255,255,255)",
              "fontFamily": "omnes",
              "fontSize": 18,
              "textAlign": "left",
            },
            Object {
              "fontFamily": "montserrat",
              "fontSize": 16,
              "fontWeight": "800",
            },
            Object {
              "marginTop": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        WHY FACEBOOK?
      </Text>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(255,255,255)",
                "fontFamily": "omnes",
                "fontSize": 18,
                "textAlign": "left",
              },
              Object {
                "fontFamily": "montserrat",
                "fontSize": 16,
                "fontWeight": "800",
              },
              Object {},
            ]
          }
        >
          USER PRIVACY
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(255,255,255)",
                "fontFamily": "omnes",
                "fontSize": 18,
                "textAlign": "left",
              },
              Object {},
            ]
          }
        >
          By signing up with Facebook, we can block your profile from your Facebook friends.
        </Text>
      </View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(255,255,255)",
                "fontFamily": "omnes",
                "fontSize": 18,
                "textAlign": "left",
              },
              Object {
                "fontFamily": "montserrat",
                "fontSize": 16,
                "fontWeight": "800",
              },
              Object {},
            ]
          }
        >
          NO FAKE USERS
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(255,255,255)",
                "fontFamily": "omnes",
                "fontSize": 18,
                "textAlign": "left",
              },
              Object {},
            ]
          }
        >
          Facebook does a pretty good job at purging fake users. Signing up with Facebook lowers the chances of fake users joining Trippple.
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgb(86,96,103)",
            "borderRadius": 50,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={1}
          style={
            Object {
              "height": 15,
              "width": 15,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
